Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -85,6 +88,23 @@ const CodeEditor: React.FC<CodeEditorProps> = (props) => {
     setLgEditor(lgEditor);
   };
 
+  React.useEffect(() => {
+    const abortController = new AbortController();
+    (async () => {
+      try {
+        const variables = await getMemoryVariables(projectId, { signal: abortController.signal });
+        setMemoryVariables(variables);
+      } catch (e) {
+        // error can be due to abort
+      }
+    })();
+
+    // clean up pending async request
+    () => {
```

Comment:
Don't you want to return this function from the effect so it is the cleanup function?
Consider if it would be better to clean up the abortController in a finally block.

Reply:
oh, nice catch.